packages/imgareaselect/src/index.js

```diff
@@ -0,0 +1,1241 @@
+       */
+      setTimeout(function () { options.onInit(img, getSelection()); }, 0);
+    }
+
+    /**
+     * Document keypress event handler
+     *
+     * @param event
+     *            The event object
+     * @return false
+     */
+    var docKeyPress = function(event) {
+      var k = options.keys, d, t, key = event.keyCode;
+
+      d = !isNaN(k.alt) && (event.altKey || event.originalEvent.altKey) ? k.alt :
+        !isNaN(k.ctrl) && event.ctrlKey ? k.ctrl :
+          !isNaN(k.shift) && event.shiftKey ? k.shift :
+            !isNaN(k.arrows) ? k.arrows : 10;
+
+      if (k.arrows == 'resize' || (k.shift == 'resize' && event.shiftKey) ||
+        (k.ctrl == 'resize' && event.ctrlKey) ||
+        (k.alt == 'resize' && (event.altKey || event.originalEvent.altKey)))
+      {
+        /* Resize selection */
+
+        switch (key) {
+          case 37:
+            /* Left */
+            d = -d;
+          case 39:
+            /* Right */
+            t = max(x1, x2);
+            x1 = min(x1, x2);
+            x2 = max(t + d, x1);
+            fixAspectRatio();
+            break;
+          case 38:
+            /* Up */
+            d = -d;
+          case 40:
+            /* Down */
+            t = max(y1, y2);
+            y1 = min(y1, y2);
+            y2 = max(t + d, y1);
+            fixAspectRatio(true);
+            break;
+          default:
+            return;
+        }
+
+        doResize();
+      }
+      else {
+        /* Move selection */
+
+        x1 = min(x1, x2);
+        y1 = min(y1, y2);
+
+        switch (key) {
+          case 37:
+            /* Left */
+            doMove(max(x1 - d, left), y1);
+            break;
+          case 38:
+            /* Up */
+            doMove(x1, max(y1 - d, top));
+            break;
+          case 39:
+            /* Right */
+            doMove(x1 + min(d, imgWidth - selX(x2)), y1);
+            break;
+          case 40:
+            /* Down */
+            doMove(x1, y1 + min(d, imgHeight - selY(y2)));
+            break;
+          default:
+            return;
+        }
+      }
+
+      return false;
+    };
+
+    /**
+     * Apply style options to plugin element (or multiple elements)
+     *
+     * @param $elem
+     *            A jQuery object representing the element(s) to style
+     * @param props
+     *            An object that maps option names to corresponding CSS
+     *            properties
+     */
+    function styleOptions($elem, props) {
+      for (var option in props)
+        if (options[option] !== undefined)
+          $elem.css(props[option], options[option]);
+    }
+
+    /**
+     * Set plugin options
+     *
+     * @param newOptions
+     *            The new options object
+     */
+    function setOptions(newOptions) {
+      if (newOptions.parent) {
+        // Patched XSS vulnerability.
+        ( $parent = $.find( newOptions.parent ) ).append( $box.add( $outer ) );
```

Comment:
@swissspidy This line seems to be the problem on QA env. 
```
imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1 Uncaught TypeError: (intermediate value)(intermediate value).append is not a function
    at Ce (imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1)
    at ke (imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1)
    at new e.imgAreaSelect (imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1)
    at HTMLImageElement.<anonymous> (imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1)
    at Function.each (load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils&ver=5.7.2:2)
    at s.fn.init.each (load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils&ver=5.7.2:2)
    at s.fn.init.e.fn.imgAreaSelect (imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1)
    at n.onImageLoad (media-views.min.js?ver=5.7.2:2)
    at HTMLImageElement.dispatch (load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils&ver=5.7.2:2)
    at HTMLImageElement.v.handle (load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils&ver=5.7.2:2)
Ce @ imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1
ke @ imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1
e.imgAreaSelect @ imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1
(anonymous) @ imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1
each @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils&ver=5.7.2:2
each @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils&ver=5.7.2:2
e.fn.imgAreaSelect @ imgareaselect.js?ver=eaa352ba5401927608167fef838cec8a:1
onImageLoad @ media-views.min.js?ver=5.7.2:2
dispatch @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils&ver=5.7.2:2
v.handle @ load-scripts.php?c=1&load[chunk_0]=jquery-core,jquery-migrate,utils&ver=5.7.2:2
```

I think we should revert this change and either leave it as it is or fix in another PR.

Reply:
Couldn't replicate this locally

Reply:
Looks like `newOptions.parent` can be a jQuery object, so we can write it like this:

```js
if (newOptions.parent) {
  $parent = newOptions.parent instanceOf $ ? newOptions.parent : $.find( newOptions.parent );
  $parent.append( $box.add( $outer ) );
}
```

Let's try that first.

Reply:
Of course if this doesn't work, we can revert it.

Reply:
Where was the original report for this xss? I personally can't see the issue, the value submitted to the selector is a node not a string selector.

Reply:
> Looks like `newOptions.parent` can be a jQuery object, so we can write it like this:
> 
> ```js
> if (newOptions.parent) {
>   $parent = newOptions.parent instanceOf $ ? newOptions.parent : $.find( newOptions.parent );
>   $parent.append( $box.add( $outer ) );
> }
> ```
> 
This didn't work. 

```
media-views.js?ver=5.8:6990 Uncaught TypeError: this.$image.imgAreaSelect is not a function
    at n.onImageLoad (media-views.js?ver=5.8:6990)
    at Zn (underscore.min.js?ver=1.8.3:2)
    at HTMLImageElement.<anonymous> (underscore.min.js?ver=1.8.3:2)
    at HTMLImageElement.<anonymous> (underscore.min.js?ver=1.8.3:2)
    at HTMLImageElement.dispatch (jquery.js?ver=3.6.0:5430)
    at HTMLImageElement.elemData.handle (jquery.js?ver=3.6.0:5234)
onImageLoad @ media-views.js?ver=5.8:6990
Zn @ underscore.min.js?ver=1.8.3:2
(anonymous) @ underscore.min.js?ver=1.8.3:2
(anonymous) @ underscore.min.js?ver=1.8.3:2
dispatch @ jquery.js?ver=3.6.0:5430
elemData.handle @ jquery.js?ver=3.6.0:5234
```
This does work 
```
$parent = newOptions.parent.append( $box.add( $outer ) );
```

Reply:
> Where was the original report for this xss? I personally can't see the issue, the value submitted to the selector is a node not a string selector.

See https://github.com/google/web-stories-wp/pull/8501#discussion_r679007074

It's GitHub security check. Only visible to admins.

And it's only a _potential_ vulnerability known to jQuery plugins in general, where the options are passed to `$()` unsanitized.